### DISCUSSION TITLE
CMAKE_BUILD_TYPE forced to Release

### DISCUSSION THREAD

**Discussion by redtide** [tsujan/Arqiver#98] [Q&A] [answered]
Is there some issue by using a Debug build to force the Release build from CMakeFiles.txt?

**Comment by tsujan** [comment]
I didn't understand the question. Could you please be more specific?

**Comment by redtide** [comment]
I opened the CMakeLists.txt in QtCreator as project, then I selected the Debug configuration and an error said that only Release build is available, though Debug was choosen at first configuration.
Then I looked in the [CMakeLists.txt](https://github.com/tsujan/Arqiver/blob/master/CMakeLists.txt#L3) that it was set there, forcing Release as build configuration.
Then I replaced it with a
```cmake
if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE Release)
endif()
```
so that the Debug was available when selected in QtC (or via command line).
At that point I was wondering if that was forced because some problem when using Debug, and I asked here.

**Reply by tsujan** [reply, marked as answer]
Oh, now I understand your question! You meant I might have forced the release build because I saw some issue with the debug build. The answer is no, I didn't. I just prefer the Release build and have a habit to do so. Those who want a debug build will find their way, I think.

**Reply by redtide** [reply]
I don't see why not apply the change I did then, you still have Release as default and I can use Debug without have to modify the CMake code.

**Reply by redtide** [reply]
I forgot: that might help also Arch packagers, which prefer to set explicitly the build type to [None](https://wiki.archlinux.org/title/CMake_package_guidelines#Fixing_the_automatic_optimization_flag_override) (like [here](https://gitlab.archlinux.org/archlinux/packaging/packages/lxqt-archiver/-/blob/main/PKGBUILD#L25)).

**Reply by tsujan** [reply]
> that might help also Arch packagers

I'll consider the change you proposed. Thanks!
